contrib/ruff.toml

```diff
@@ -7,6 +7,8 @@ extend-ignore = [
     # https://github.com/charliermarsh/ruff#flake8-type-checking-tch
     # This is likely to be a high-touch rule that most contribs don't need to care about.
     "TCH001",
+    # https://beta.ruff.rs/docs/rules/#flake8-use-pathlib-pth
+    "PTH",
```

Comment:
@austiezr we are adding some new rules regarding using `pathlib` instead of `os.path`.
There were a lot of violations in `contrib` initially so I opted to disable this rule initially for contrib.
So I'll leave it up to you and your team whether or not you want to adopt this as well.